components/mpas-albany-landice/src/shared/mpas_li_time_average_coupled.F

```diff
@@ -0,0 +1,137 @@
+        call mpas_pool_get_array(timeAveragingPool, 'avgFaceMeltingThickness', avgFaceMeltingThickness)
+
+        do iCell = 1, nCells
+           avgBareIceAblation(iCell) = 0.0_RKIND
+           avgCalvingThickness(iCell) = 0.0_RKIND
+           avgFaceMeltingThickness(iCell) = 0.0_RKIND
+        end do
+
+        timeAccumulatedCoupled = 0.0_RKIND
+
+    end subroutine li_time_average_coupled_init
+
+!***********************************************************************
+!
+!  routine li_time_average_coupled_accumulate
+!
+!> \brief   Coupled time averager accumulation
+!> \author  Stephen Price
+!> \date    04 March 2025
+!> \details
+!>  This routine accumulated the coupled time averaging fields
+!
+!-----------------------------------------------------------------------
+    subroutine li_time_average_coupled_accumulate(timeAveragingPool, geometryPool, meshPool)
+            
+        use li_setup 
+
+        type (mpas_pool_type), intent(inout) :: timeAveragingPool
+        type (mpas_pool_type), intent(in) :: geometryPool
+        type (mpas_pool_type), intent(in) :: meshPool
+
+        real (kind=RKIND), dimension(:), pointer :: bareIceAblation, avgBareIceAblation, &
+                                                    calvingThickness, avgCalvingThickness, &
+                                                    faceMeltingThickness, avgFaceMeltingThickness
+        integer, pointer :: nCells
+
+        real (kind=RKIND), pointer :: timeAccumulatedCoupled, rhoi, deltat
+
+        integer :: iCell
+
+        call mpas_pool_get_dimension(meshPool, 'nCells', nCells)
+        call mpas_pool_get_config(liConfigs, 'config_ice_density', rhoi)
+        call mpas_pool_get_array(meshPool, 'deltat', deltat)
+        call mpas_pool_get_array(timeAveragingPool, 'timeAccumulatedCoupled', timeAccumulatedCoupled)
+        call mpas_pool_get_array(geometryPool, 'bareIceAblation', bareIceAblation)
+        call mpas_pool_get_array(geometryPool, 'calvingThickness', calvingThickness)
+        call mpas_pool_get_array(geometryPool, 'faceMeltingThickness', faceMeltingThickness)
+        call mpas_pool_get_array(timeAveragingPool, 'avgBareIceAblation', avgBareIceAblation)
+        call mpas_pool_get_array(timeAveragingPool, 'avgCalvingThickness', avgCalvingThickness)
+        call mpas_pool_get_array(timeAveragingPool, 'avgFaceMeltingThickness', avgFaceMeltingThickness)
+
+        do iCell = 1, nCells
+
+           avgBareIceAblation(iCell) = ( avgBareIceAblation(iCell) * timeAccumulatedCoupled &
+                                               + bareIceAblation(iCell) * deltat ) / ( timeAccumulatedCoupled + deltat )
+
+           avgCalvingThickness(iCell) = ( avgCalvingThickness(iCell) * timeAccumulatedCoupled &
+                                                + calvingThickness(iCell) * deltat ) / ( timeAccumulatedCoupled + deltat ) &
+                                                * rhoi / deltat
```

Comment:
This term should only be applied to the new increment being accumulated

Reply:
I've corrected this as:

`avgFaceMeltingThickness(iCell) = ( avgFaceMeltingThickness(iCell) * timeAccumulatedCoupled + faceMeltingThickness(iCell) * deltat * (rhoi / deltat) ) / (timeAccumulatedCoupled + deltat )`

Note that I'm leaving in the two "deltat" terms rather than cancelling them out, ideally to be more clear about the reason the "rhoi/deltat" term is there (for unit conversion).

Reply:
What you wrote in the comment looks correct to me.  And I agree that I prefer to have the two deltat instances for clarity (or alternatively an explanation in a comment).  Note that the diff I see does *not* show the equation your wrote in the comment, so maybe you have unpushed commits?

Reply:
I had not pushed yet at the point you made this comment. The code on this branch should be updated now to reflect the changes from this discussion.